### PR TITLE
Cloudwatch event

### PR DESCRIPTION
**Pull request by willbengtson**
Add CloudWatch Events to CloudAux.

### CODE REVIEW

File: cloudaux/orchestration/aws/events.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            account=conn.get('account'),
+            name=rule_name),
+        'name': rule_name,
+        'region': conn.get('region'),
+        '_version': 1
+    }
+
+
+@modify_output
+def get_event(rule_name, flags=FLAGS.ALL, **conn):
```
Comment (monkeysecurity) [inline cloudaux/orchestration/aws/events.py]: Would be nice if get_event could take a rule_name, an ARN, or the dictionary returned by list_rules.

File: cloudaux/orchestration/aws/events.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+from cloudaux.aws.events import list_targets_by_rule
+from cloudaux.decorators import modify_output
+from flagpole import FlagRegistry, Flags
+
+from botocore.exceptions import ClientError
+import logging
+import json
+
+
+logger = logging.getLogger('cloudaux')
```
Comment (monkeysecurity) [inline cloudaux/orchestration/aws/events.py]: unused

File: cloudaux/orchestration/aws/events.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
+from cloudaux.aws.events import describe_rule
+from cloudaux.aws.events import list_targets_by_rule
+from cloudaux.decorators import modify_output
+from flagpole import FlagRegistry, Flags
+
+from botocore.exceptions import ClientError
+import logging
+import json
```
Comment (monkeysecurity) [inline cloudaux/orchestration/aws/events.py]: unused

File: cloudaux/orchestration/aws/events.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
+from cloudaux.aws.events import describe_rule
+from cloudaux.aws.events import list_targets_by_rule
+from cloudaux.decorators import modify_output
+from flagpole import FlagRegistry, Flags
+
+from botocore.exceptions import ClientError
+import logging
```
Comment (monkeysecurity) [inline cloudaux/orchestration/aws/events.py]: unused

File: cloudaux/orchestration/aws/events.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
+from cloudaux.aws.events import describe_rule
+from cloudaux.aws.events import list_targets_by_rule
+from cloudaux.decorators import modify_output
+from flagpole import FlagRegistry, Flags
+
+from botocore.exceptions import ClientError
```
Comment (monkeysecurity) [inline cloudaux/orchestration/aws/events.py]: unused

File: cloudaux/aws/events_bus.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+from cloudaux.aws.sts import sts_conn
+from cloudaux.aws.decorators import rate_limited, paginated
+from botocore.exceptions import ClientError
```
Comment (monkeysecurity) [inline cloudaux/aws/events_bus.py]: unused

File: cloudaux/aws/events_bus.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
+from cloudaux.aws.sts import sts_conn
+from cloudaux.aws.decorators import rate_limited, paginated
```
Comment (monkeysecurity) [inline cloudaux/aws/events_bus.py]: paginated is unused

File: cloudaux/aws/events.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+@paginated('Rules', request_pagination_marker="NextToken",
+           response_pagination_marker="NextToken")
+@rate_limited()
+def list_rules(client=None, **kwargs):
+    """
+    NamePrefix='string'
+    """
+    result = client.list_rules(**kwargs)
+    if not result.get("Rules"):
+        result.update({"Rules": []})
```
Comment (monkeysecurity) [inline cloudaux/aws/events.py]: Does this happen where the result doesn't have a `Rules` key?

File: cloudaux/orchestration/aws/events.py (hunk 1 above)
Comment (willbengtson) [reply cloudaux/orchestration/aws/events.py]: 👍 will add

File: cloudaux/orchestration/aws/events.py [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+    return {
+        'description': rule['Description'],
+        'state': rule['State'],
+        'rule': rule_detail
+    }
+
+
+@registry.register(flag=FLAGS.BASE)
+def get_base(rule_name, **conn):
```
Comment (monkeysecurity) [inline cloudaux/orchestration/aws/events.py]: Seems odd that `get_base` would work even when the underlying event rule does not exist.

I think the other cloudaux orchestrators typically call the describe function from within the `get_base` method, but only if the base_fields are not already set:

https://github.com/Netflix-Skunkworks/cloudaux/blob/master/cloudaux/orchestration/aws/elbv2.py#L57

File: cloudaux/__about__.py [hunk 10]
```diff
@@ -13,10 +13,10 @@
 __summary__ = 'Cloud Auxiliary is a python wrapper and orchestration module for interacting with cloud providers'
 __uri__ = 'https://github.com/Netflix-Skunkworks/cloudaux'
 
-__version__ = '1.3.5'
+__version__ = '1.3.8'
```
Comment (monkeysecurity) [inline cloudaux/__about__.py]: @mikegrima took `1.3.8` - Could you bump to `1.3.9` ?

File: cloudaux/aws/events.py (hunk 8 above)
Comment (monkeysecurity) [reply cloudaux/aws/events.py]: Do you have any idea what I meant by my previous comment? ^^

**Comment by monkeysecurity** [comment]
Were you able to review this with @mcpeak ?

File: cloudaux/__about__.py (hunk 10 above)
Comment (monkeysecurity) [reply cloudaux/__about__.py]: (Actually, I'll make the change.)

**Pull request merged**